### PR TITLE
[PT] Translate Types and Specifications

### PR DESCRIPTION
**Pull request by vicdda**
Hi, i have done the translation of the lesson of types and specifications

### CODE REVIEW

**Comment by doomspork** [comment]
Thank you @vicdda and welcome!  @pragmaticivan would you mind taking a peek at this if you've got time?

**Comment by vicdda** [comment]
Thanks !!

Em 30/03/2017 19:02, "Sean Callan" <notifications@github.com> escreveu:

File: pt/lessons/advanced/typespec.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+Nessa lição vamos aprender sobre a sintaxe de `@spec` e `@type`. O primeiro é mais uma sintaxe complementar para escrever documentação que pode ser analisada por ferramentas. A segunda nos ajuda a escrever código de fácil leitura e entendimento.
+
+{% include toc.html %}
+
+## Introdução
+
+Não é incomum querer descrever a interface de sua função. Claro, você pode utilizar [@doc annotation](../../basics/documentation), mas isso é somente informação para outros desenvolvedores. Isso não será chegado em tempo de compilação. Para isso, Elixir tem uma anotação chamada `@spec` para descrever especificação de função que vai ser analisada pelo compilador.
+
+Em alguns casos, a especificação é grande e complicada. Se você quiser reduzir a complexidade, você deverá introduzir um tipo de definição personalizada. Elixir tem a anotação `@type` para isso. Em contar partida, Elixir é uma linguagem dinâmica. Isso significa que toda informação a respeito do tipo será ignorado pelo compilador, mas pode ser utilizada por outras ferramentas.
```
Comment (pragmaticivan) [inline pt/lessons/advanced/typespec.md]: `Em contra partida`..

**Comment by vicdda** [comment]
Done !

**Comment by pragmaticivan** [comment]
Thanks @vicdda !

**Pull request merged**